exercises/exercise-list.md

```diff
@@ -0,0 +1,278 @@
+# Java For Beginners - Exercises
+
+* [Topic 2 Exercises](#exercise-list-topic-2)
+* [Topic 3 Exercises](#exercise-list-topic-3)
+
+# Exercise List Topic 2
+
+This exercise list is to practice the content presented in [lecture 2](../lessons/lesson2)
+
+## \[VABO] Variables and Basic Operators
+### Printing Varaibles
+In a `main`, define a `String`, `Integer`, and `Boolean` variable.
+1. `System.out.println` (print) each of them separately.
+2. Print them together in a sentence (which may or may not make sense).
+
+### \[MAOP] Math Operations
+Write a program that has 2 Integer variables `a` and `b`, print the result of each of applying addition, subtraction, division, multiplication, and modulus of `a` to `b`.
+
+e.g given a = 10 and b = 2, print:
+
+```java
+//print 12 (a+b)
+//print 8 (a-b)
+//etc
+```
+
+### \[MCMP] Math Comparisons
+Write a program that has 2 Integer variables `a` and `b`, print the result of each of applying equal, greater than, less than, greater than or equal, and less than or equal of `a` to `b`.
+
+e.g given a = 10 and b = 2, print:
+```java
+a equals b // false
+a > b // true
+a < b // false
+```
+
+### \[HINH] Hello, *Insert name here*!
+Write your own "Hello, world!" program, then extend it to take a name as input from the console and say hello to the inputted name instead.
+
+You should use `UserInputUtil` in the `eca.util` package, you will have to `import` it.
+
+### \[OOEV] OddOrEven
+Write a Java program to accept a number and check the number is even or not. Output the `Boolean` value of  the result
+
+e.g for a number a = 10 print `true`
+
+### \[SETE] Sum is equal to Expected
+Write a Java program to calculate the sum of two integers and return true if the sum is equal to a third integer
+
+e.g for:
+```java
+a = 10
+b = 5
+c = 15
+print true (a + b = c)
+```
+
+### \[GWNT] Guess What Number I'm Thinking
+Write a program that thinks of a random number between 1 and 10.
+Let the user enter a number, if it is correct print "You win!" otherwise print "You lose!".
+
+You will need some way to get a random number within a range, [java has many ways and developers are still arguing over which are the best](https://stackoverflow.com/questions/363681/how-do-i-generate-random-integers-within-a-specific-range-in-java).
```

Comment:
I'd argue that our students don't really care about whether this is the best way of getting Random or not. I would probably simplify this to say "you can get a random number by using: `new Random().nextInt(x)` changing x to be the upper bound of your range"

Reply:
yep removed